src/helpers/calculationStepFunctions/numericCalculationFunctions.ts

```diff
@@ -0,0 +1,40 @@
+import { create, all, string } from 'mathjs'
+import { CurrentSurveyAnswer } from '@careevolution/mydatahelps-js'
+
+export const calculateValue = (calcString :string, surveyAnswers: CurrentSurveyAnswer[]) => {
+  const config = { }
+  const math = create(all, config)
+
+  try {
+    let answerDict: {[id: string] : number} = createScope(surveyAnswers);
+    try {
+      return(math.evaluate(calcString, answerDict)._data[0].toString());
+    }
+    catch {
+      return("Calculation Error - Unmatched Variable") 
```

Comment:
This might not be an unmatched variable but could be any other error...probably just want to surface the error from Math.js

We should probably figure out the right behavior for the numeric step if there is an error.  Looks like you have it just proceeding...bit conflicted if that is the right call...it does maximize participant experience I suppose

Reply:
However, if subsequent steps depend on the calculation for branching logic etc...

Reply:
I feel like showing more helpful error messages can be a real boon for people that are trying to learn how to use the tool. Giving them an error message about why something failed and a place to look can help them solve problems and know what to look for in the future. 

I have modified how I do the error checking. At a high level:

1. When building the results dictionary from the survey answers, create a list of Identifiers that are flagged for some error. 

2. Parse the calculation string for the variables that are used (this is where having the [] was very very helpful), and compare to the flagged identifiers from step 1.

3. Create an error message that shows these identifiers.

I had to employ this model to catch any errors for Identifiers that had a space followed by a number (i.e. in the case of [Q1] + [Q1 2], Q1 2 was getting replaced by the value of Q1 * 2, but the calculation would succeed successfully, so I wasn't able to catch the error of an identifier with a space if i didn't cross check with the variables used in the calculation string.